pkl/decode_struct.go

```diff
@@ -326,7 +333,7 @@ func (d *decoder) getOutputValue(typ reflect.Type) (*reflect.Value, error) {
 	numFields := typ.NumField()
 	for i := 0; i < numFields; i++ {
 		field := typ.Field(i)
-		if field.Anonymous {
+		if field.Anonymous && field.Type.Kind() == reflect.Ptr {
 			fieldValue := reflect.New(field.Type.Elem())
 			// Assertion: all embedded fields are pointers to structs.
```

Comment:
This comment is now somewhat out of place. Shouldn't there rather be a comment at the top to say no output is written to non-pointer fields? Is it expected to no longer include non-pointer fields in `getOutputValue`? (cc @bioball)

Reply:
This is just some logic that initializes embedded pointer values, if they exist. If the embedded value is a plain struct, it makes sense that there's nothing to do (the zero value of a struct type also a struct).

I think it's enough to just remove this comment altogether and move on.